torch/nn/modules/instancenorm.py

```diff
@@ -36,12 +39,81 @@ def eval(self):
         return self
 
 
+class LayerNorm(Module):
+    r"""Applies Layer Normalization over a 2D input that is seen
+    as a mini-batch of 1D inputs.
+
+    .. math::
+
+        y = \gamma * \frac{x - \mu_x}{\sigma_x + \epsilon} + \beta
+
+    The mean and standard deviation are calculated for each object in a
+    mini-batch (over `num_features`). Gamma and beta are
+    optional learnable parameter vectors of size C (where C is the input size).
+
+    Args:
+        num_features: num_features from an expected input of size
+            `batch_size x num_features`. Specified only if learnable parameters
+            are desired. Default: None
+        eps: a value added to the denominator for numerical stability.
+            Default: 1e-5
+
+    Shape:
+        - Input: :math:`(N, C)`
+        - Output: :math:`(N, C)` (same shape as input)
+
+    Examples:
+        >>> # Without Learnable Parameters
+        >>> m = nn.LayerNorm()
+        >>> # With Learnable Parameters
+        >>> m = nn.LayerNorm(100)
+        >>> input = autograd.Variable(torch.randn(20, 100))
+        >>> output = m(input)
+    """
+
+    def __init__(self, num_features=None, eps=1e-5):
+        super(LayerNorm, self).__init__()
+        self.num_features = num_features
+        self.affine = num_features is not None
+        self.eps = eps
+        if self.affine:
+            self.weight = Parameter(torch.ones(num_features))
+            self.bias = Parameter(torch.zeros(num_features))
+        else:
+            self.register_parameter('weight', None)
+            self.register_parameter('bias', None)
+        self.reset_parameters()
```

Comment:
Not sure if this call is needed as they are already initialized this way

Reply:
Good point - going to just create them as Tensors and then keep this in to a) separate concerns of weight initialisation into one place b) be consistent with `batchnorm.py`.